File: Problem5_Tokyo/5_claude_sol1.py
import sys


def main():
    data = sys.stdin.read().split()
    idx = 0

    T = int(data[idx])
    idx += 1
    out = []

    for _ in range(T):
        N = int(data[idx])
        idx += 1

        prefix = [0] * (N + 1)
        for i in range(1, N + 1):
            c = data[idx]
            x = int(data[idx + 1])
            idx += 2
            delta = x if c == 'L' else -x
            prefix[i] = prefix[i - 1] + delta

        first_occurrence = {0: 0}
        best_len = 0
        best_s = 0
        best_e = 0

        for i in range(1, N + 1):
            p = prefix[i]
            if p in first_occurrence:
                length = i - first_occurrence[p]
                if length > best_len:
                    best_len = length
                    best_s = first_occurrence[p] + 1
                    best_e = i
            else:
                first_occurrence[p] = i

        if best_len == 0:
            out.append("0 0 0")
        else:
            out.append(f"{best_len} {best_s} {best_e}")

    sys.stdout.write("\n".join(out) + "\n")


if __name__ == "__main__":
    main()

File: Problem5_Tokyo/5_claude_sol1.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int T = nextInt(in);
        StringBuilder sb = new StringBuilder();

        while (T-- > 0) {
            int N = nextInt(in);
            long[] prefix = new long[N + 1];

            for (int i = 1; i <= N; i++) {
                char dir = nextNonSpaceChar(in);
                long x = nextInt(in);
                long delta = (dir == 'L') ? x : -x;
                prefix[i] = prefix[i - 1] + delta;
            }

            HashMap<Long, Integer> firstOcc = new HashMap<>();
            firstOcc.put(0L, 0);

            long bestLen = 0;
            int bestS = 0, bestE = 0;

            for (int i = 1; i <= N; i++) {
                Integer occ = firstOcc.get(prefix[i]);
                if (occ == null) {
                    firstOcc.put(prefix[i], i);
                } else {
                    long len = i - occ;
                    if (len > bestLen) {
                        bestLen = len;
                        bestS = occ + 1;
                        bestE = i;
                    }
                }
            }

            if (bestLen == 0) {
                sb.append("0 0 0\n");
            } else {
                sb.append(bestLen).append(' ').append(bestS).append(' ').append(bestE).append('\n');
            }
        }

        System.out.print(sb);
    }

    private static int nextInt(DataInputStream in) throws IOException {
        int ret = 0;
        int b = in.read();
        while (b < '0' || b > '9') {
            if (b == -1)
                return -1;
            b = in.read();
        }
        while (b >= '0' && b <= '9') {
            ret = ret * 10 + (b - '0');
            b = in.read();
        }
        return ret;
    }

    private static char nextNonSpaceChar(DataInputStream in) throws IOException {
        int b = in.read();
        while (b == ' ' || b == '\n' || b == '\r' || b == '\t') {
            b = in.read();
        }
        return (char) b;
    }
}